spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarProducerFactory.java

```diff
@@ -60,10 +63,16 @@ public Producer<T> createProducer(String topic, Schema<T> schema) throws PulsarC
 	@Override
 	public Producer<T> createProducer(String topic, Schema<T> schema, MessageRouter messageRouter)
 			throws PulsarClientException {
-		return doCreateProducer(topic, schema, messageRouter);
+		return createProducer(topic, schema, messageRouter, null);
```

Comment:
**[UNRELATED_NIT]** So this is something I introduced and did catch until now... 

When I use the `"N 'action' methods delegate to the 'doAction' impl method"` pattern (I made that term up just now 😉 ) I typically try to avoid the N level method chaining and just have each call to the `doAction` method directly. The main reason is it keeps the top-level "action methods" API decoupled a bit and makes it easier to evolve one of the apis w/o worrying about breaking the other API as well (aka smaller blast radius).

As it is currently, any change to `createProducer(String topic, Schema<T> schema, MessageRouter messageRouter, List<ProducerInterceptor> producerInterceptors)` would be considered a possible change to `createProducer(String topic, Schema<T> schema, MessageRouter messageRouter)` which in turn would bring `createProducer(String topic, Schema<T> schema)` into the potential blast radius. 

Albeit, the current impl has no logic so it would not be noticeable. However, if we start varying them by adding validation in each method then it could bite us later. 

ℹ️ Also note that we are doing the same in PulsarOperations/Template (just noticed this).

Reply:
This is a good point, I will also create a separate PR to change the PulsarOperations & -Template in this regard